### PR TITLE
feat: add esm support [breaking change]

### PR DESCRIPTION
**Pull request by anonrig**
### Breaking Changes

- Converted app to ESM

## Minor changes

- Move `grpc-create-metadata` to inside the main repo
  - It was causing inconsistencies since the package was also dependant to `@grpc/grpc-js`
- Replace `standard` with `eslint`
- Remove `p-map`
- Remove `is-stream`
- Remove `ascallback`
- Remove `standard`
- Remove `is-CI`
- Remove `lodash.find`
- Remove `lodash.intersection`
- Remove `lodash.upperfirst`
- Remove `lodash.values`
- Remove `error-inject`

### CODE REVIEW

**Comment by anonrig** [comment]
@bojand I've been working on a particular test which is now failing. can you take a look into it?

File: .github/workflows/nodejs.yml [hunk 1]
```diff
@@ -3,48 +3,47 @@ name: Node CI
...
 
 jobs:
   build:
-
     runs-on: ubuntu-latest
 
     strategy:
       matrix:
-        node-version: [14.x]
+        node-version: [14, 15, 16]
```
Comment (bojand) [inline .github/workflows/nodejs.yml:17]: we don't really need 15 here, as odd-numbered releases become unsupported after 6 months

**Comment by bojand** [comment]
Thank you for the PR. I think it might be more useful, and easier to review if we break this down into smaller PRs, starting with the removal, and cleanup of some of the dependencies which seems useful.

What's the purpose of switching it ES module? Are there any tangible benefits?

As for style change, I would prefer keeping standard as is.

File: .github/workflows/nodejs.yml (hunk 1 above)
Comment (anonrig) [reply .github/workflows/nodejs.yml:17]: Correct,I just wanted to make sure we're 100% covered.

**Comment by anonrig** [comment]
Thanks @bojand I'll open a new pull request for removing the dependencies. ESM is the next step of JS evolution where the syntax of JS is merged with FE and BE applications. My main reasoning was based upon

- Initially, `p-map` package rewrote their package in ESM, which caused me to make this change, but after working on it for a couple of days, I saw that `p-map` package is not needed, so I removed it. My initial justification is not valid anymore, so... (refer to https://gist.github.com/sindresorhus/a39789f98801d908bbc7ff3ecc99d99c)
- EOL of Node 10. ESM is natively supported by Node.js 12 and later.
- Supports top-level await by default.

This will provide you a clearer explaination for the differences between ESM and commonjs.
https://nodejs.org/api/esm.html#esm_interoperability_with_commonjs

**Pull request closed without merging**